src/alignfunctor.cpp

```diff
@@ -653,9 +653,13 @@ FunctorCode AlignVerticallyFunctor::VisitStaff(Staff *staff)
         Verse *verse = vrv_cast<Verse *>((*sylIterator)->GetFirstAncestor(VERSE));
         if (verse) {
             const int verseNumber = verse->GetN();
+            const data_STAFFREL versePlace = verse->GetPlace();
             const bool verseCollapse = m_doc->GetOptions()->m_lyricVerseCollapse.GetValue();
-            if (!alignment->GetVersePosition(verseNumber, verseCollapse)) {
-                alignment->AddVerseN(verseNumber);
+            if ((versePlace == STAFFREL_above) && !alignment->GetVersePositionAbove(verseNumber, verseCollapse)) {
+                alignment->AddVerseN(verseNumber, verse->GetPlace());
+            }
+            if ((versePlace != STAFFREL_above) && !alignment->GetVersePositionBelow(verseNumber, verseCollapse)) {
```

Comment:
This looks like it's two lines that are a duplicate of each other? Is this correct? If so, then couldn't the `if` statements be combined?